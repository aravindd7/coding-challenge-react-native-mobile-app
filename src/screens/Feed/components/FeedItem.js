import React from "react";

import { Text, View, TouchableOpacity } from "react-native";

class FeedItem extends React.PureComponent {
  render() {
    const { item } = this.props;

    return (
      <View>
        <TouchableOpacity onPress={() => {}}>
          <View>
            <Text>FeedItem</Text>
          </View>
        </TouchableOpacity>
      </View>
    );
  }
}

export default FeedItem;
