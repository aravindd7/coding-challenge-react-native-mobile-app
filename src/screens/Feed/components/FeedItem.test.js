import React from "react";

import ShallowRenderer from "react-test-renderer/shallow"; // ES6

import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallow } from "enzyme";

import FeedItem from "./FeedItem";

configure({ adapter: new Adapter() });

const props = {
  id: "1234",
  title: "title",
  author: "author",
  num_comments: 3,
  score: 2,
  created_utc: 1517115254,
  thumbnail: "thumbnail.jpg",
  url: "http://example.com",
  selectItem: () => {}
};

it("match on Shallow mode", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<FeedItem {...props} />);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});

it("match call selectItemMock", () => {
  const selectItemMock = jest.fn();

  const wrapper = shallow(<FeedItem {...props} selectItem={selectItemMock} />);
  wrapper.instance()._selectItem();

  expect(selectItemMock.mock.calls).toMatchSnapshot();
});
