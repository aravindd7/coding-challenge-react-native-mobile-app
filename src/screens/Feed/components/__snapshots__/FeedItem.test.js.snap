// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match call selectItemMock 1`] = `
Array [
  Array [
    Object {
      "title": "title",
      "url": "http://example.com",
    },
  ],
]
`;

exports[`match on Shallow mode 1`] = `
<View>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "marginBottom": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 2,
            "padding": 5,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "thumbnail.jpg",
            }
          }
          style={
            Object {
              "flex": 1,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 7,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 11,
              "margin": 3,
              "textAlign": "right",
            }
          }
        >
          <TimeAgo
            hideAgo={false}
            interval={60000}
            time={undefined}
          />
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontWeight": "bold",
              "margin": 3,
            }
          }
        >
          title
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
              "margin": 3,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "flex": 4,
                "fontSize": 10,
                "margin": 3,
              }
            }
          >
            author
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "flex": 3,
                "fontSize": 10,
                "margin": 3,
              }
            }
          >
            Score: 
            2
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "flex": 3,
                "fontSize": 10,
                "margin": 3,
              }
            }
          >
            3
             comments
          </Text>
        </View>
      </View>
    </View>
  </TouchableOpacity>
</View>
`;
